plane/src/proxy/proxy_service.rs

```diff
@@ -324,26 +337,15 @@ impl RequestHandler {
                 .parse()
                 .expect("Backend ID is a valid header value."),
         );
-        headers.insert(
-            "Access-Control-Allow-Origin",
-            "*".parse().expect("Valid header value."),
-        );
-        headers.insert(
-            "Access-Control-Allow-Methods",
-            "GET, POST, PUT, DELETE, OPTIONS"
-                .parse()
-                .expect("Valid header value."),
-        );
-        headers.insert(
-            "Access-Control-Allow-Headers",
-            "Content-Type, Authorization"
-                .parse()
-                .expect("Valid header value."),
-        );
-        headers.insert(
-            "Access-Control-Allow-Credentials",
-            "true".parse().expect("Valid header value."),
-        );
+
+        for (key, value) in DEFAULT_CORS_HEADERS {
+            if !headers.contains_key(*key) {
```

Comment:
Note that `key` here is received as an [`AsHeaderName`](https://docs.rs/http/latest/http/header/trait.AsHeaderName.html) implementation, so it is case-normalized before it is looked up in the `HeaderMap`. So we will do the right thing even if the header name from upstream has a different casing.